<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
<html>
  <head>
    <title>
      Clump
    </title>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
    <style type="text/css">

    body {
      background: #00C8D7;
      background-image: url('bg.png');
      margin: 0;
    }
    #container {
      width: 620px;
      margin: 0 auto;
      font-family: "Helvetica Neue";
      background: #fff;
      padding: 20px;
      border-right: 6px solid #547272;
      border-left: 6px solid #547272;
    }
    h1 {
      font-size: 90px;
      margin: 0;
      letter-spacing: -.15em;
      text-shadow: #a0a 0px 0px 10px;

    }
    h1, h2 {
      font-family: "Helvetica Neue";
    }
    h2 {

      font-weight: 200;
    }
    a {
      color: #a0a;
      text-decoration: none;
    }
    .ribbon {
      background-color: #a0a;
      overflow: hidden;
      /* top left corner */
      position: absolute;
      right: -3em;
      top: 2.5em;
      /* 45 deg ccw rotation */
      -moz-transform: rotate(45deg);
      -webkit-transform: rotate(45deg);
      /* shadow */
      -moz-box-shadow: 0 0 1em #888;
      -webkit-box-shadow: 0 0 1em #888;
    }
    .ribbon a {
      border: 1px solid #aaa;
      color: #fff;
      display: block;
      font: bold 81.25% 'Helvetiva Neue', Helvetica, Arial, sans-serif;
      margin: 0.05em 0 0.075em 0;
      padding: 0.5em 3.5em;
      text-align: center;
      text-decoration: none;
      /* shadow */
      text-shadow: 0 0 0.5em #444;
    }

    </style>
  </head>
  <body>
    <div class="ribbon">
      <a href="http://www.github.com/audy/clump" rel="me">Fork me on GitHub</a>
    </div>
    <div id="container">
      <h1 id="clump">
        Clump
      </h1>
      <p>
        <a href="harekrishna@gmail.com">Austin G. Davis-Richardson</a>
      </p>
      <p>
        <blockquote>Clump is a term derived from the latin <em>clumpus</em>
        which means to shake one's belly fat in an amusing way before
        finding treasure.</blockquote>
        
        

      </p>
      <h2 id="goal">
        Goal
      </h2>
      <p>
        To investigate the usefulness of various clustering algorithms and
        distance metrics on a dataset made from Illumina sequencing reads and
        classified with the SEED database into ontology subsystem trees.
        </p><p>
        By usefulness I mean producing biologically interesting results.
        The output will be either a list of and/or graphical representation
        of the gene clusters that co-vary. 
      </p>
      <p>The dataset I'm using was generated from Illumina sequencing of 
        bacterial DNA in human stool samples.
      </p>
      <h2 id="distance_metric">
        Distance Metric
      </h2>
      <ul>
        <li>Δreads
        </li>
        <li>
          <em>p</em>-Value
        </li>
        <li>distance in subsystem tree (if possible)
        </li>
        <li>and both: Δreads * tree distance
        </li>
      </ul>
      <h2 id="clustering_algorithms">
        Clustering Algorithms
      </h2>
      <p>
        I plan on implementing the following:
      </p>
      <ul>
        <li>
          <a href="http://en.wikipedia.org/wiki/K-means_clustering">k-Means Clustering</a>
        </li>
        <li>
          <a href="http://en.wikipedia.org/wiki/Hierarchical_clustering">Hierarchical Clustering</a>
        </li>
        <li>
          <a href="http://en.wikipedia.org/wiki/Expectation-maximization_algorithm">Expectation Maximization</a>
        </li>
      </ul>
      <h2 id="dataset">
        Dataset
      </h2>
      <p>
        I have a dataset that looks like this:
      </p>
      <pre>
<code># 1 i+1 ..  n   Ontology
100 20  ..  90  System
50  10  ..  30  System; Subsystem 1
50  10  ..  60  System; Subsystem 2
25  1   ..  50  System; Subsystem 2; Subsubsystem 1
25  9   ..  10  System; Subsystem 2; Subsubsystem 2
</code>
</pre>
      <p>
        (The numbers are read counts similar to expression data on a microarray, notice they add up)
      </p>
      <p>
        Subsystems are generated from the <a href="http://www.theseed.org">SEED</a> database using tools created prior to this project.
      </p>
      <h2 id="methods">
        Methods
      </h2>
      <p>
        I will use the following:
      </p>
      <ul>
        <li>
          <a href="http://www.python.org">Python</a>
        </li>
        <li>
          <a href="http://pypi.python.org/pypi/Pycluster">PyCluster</a> which is a python wrapper for the <a href="http://bonsai.hgc.jp/~mdehoon/software/cluster/cluster.pdf">C Clustering Library</a> (PDF).
        </li>
        <li>Possibly <a href="http://numpy.scipy.org">Numpy</a>, <a href="http://www.scipy.org">SciPy</a> and <a href="http://matplotlib.sourceforge.net/">MatPlotLib</a>
        </li>
      </ul>
      <h2 id="relevant_papers">
        Relevant Papers
      </h2>
      <p>
        Just for starters…
      </p>
      <ul>
        <li>Hartigan, J. A.; Wong, M. A. (1979). “Algorithm AS 136: A K-Means Clustering Algorithm”. Journal of the Royal Statistical Society
        </li>
        <li>Ward, Joe H. (1963). “Hierarchical Grouping to Optimize an Objective Function”. Journal of the American Statistical Association
        </li>
        <li>Jain, MN Murty, PJ Flynn (1999). “Data clustering: a review”. ACM computing surveys
        </li>
        <li>Moon, T.K. (1996). “The Expectation Clustering Algorithm”. Signal Processing Magazine, IEEE
        </li>
      </ul>
    </div>
  </body>
</html>
